### PR TITLE
Move to docker so everyone is working with the same env

### PR DESCRIPTION
**Pull request by alexhartley**
We're already getting into issues where it's working on one machine but
not another. Moving to a docker setup should stop this.

This commit also adds delayed job for background jobs (email etc) and
fixes the setup script.

### CODE REVIEW

**Comment by alexhartley** [comment]
If we agree to use docker I'll merge it into master and push it up. Don't merge it from github please.

File: config/application.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 module HackspaceManagementSystem
   class Application < Rails::Application
+
```
Comment (mrstebo) [inline config/application.rb]: Not whitespace! 😅

File: app/models/user.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@ def work_out_diff
   private
 
   def generate_temp_password
-    self.password = SecureRandom.hex(10) if id.nil?
+    self.password = SecureRandom.hex(10) if id.nil? && User.count > 0
```
Comment (mrstebo) [inline app/models/user.rb]: Separate that User.count in to a separate method because it is not clear why it is there 🤔.

Comment (mrstebo) [inline app/models/user.rb]: Also? It is it in the scope of the pull request? If not it should probably be another PR

Comment (alexhartley) [inline app/models/user.rb]: Yeah it should be but there's only myself and Paul currently working on it
so thought I'd get away with it :smile:

Comment (alexhartley) [inline app/models/user.rb]: I don't like that method full stop really. I going to get rid of it. It's just a hack.

File: app/models/user.rb [hunk 3]
```diff
@@ -27,17 +27,23 @@ def work_out_diff
...
-    self.password = SecureRandom.hex(10) if id.nil? && User.count > 0
+    self.password = SecureRandom.hex(10)
+  end
+
+  def new_user?
+    id.nil?
+  end
+
+  def admin_exists?
+    User.count > 0
```
Comment (mrstebo) [inline app/models/user.rb]: Excellent. Are there roles for these users? Could do even more with those by doing a `User.count(role: :admin) > 0`. If could be beneficial if you want to assign multiple admins. Plus then you could even add extra roles later on e.g. `editor`, `wizard` etc.

That should be for another PR though due to the nature of the change :grin:

Comment (alexhartley) [inline app/models/user.rb]: Yes there's roles. It's just a string value so you could have as many `editor`, `wizard`, `coreteam` etc. That `admin_exists` is purely so the setup script can create the initial admin user. 

Good call on the `User.count(role: :admin) > 0`. I agree that makes more sense. I'll change it in a sec.

After that there's an option to set any role you like from the admin section. In future I would like to get rid of that callback completely. But for now I need to get it to a point where @phyushin can work with it. So as this is at the minute, am I good to merge into master @phyushin @mrstebo ?
